platform/mellanox/mlnx-platform-api/sonic_platform/component.py

```diff
@@ -514,32 +504,28 @@ def auto_update_firmware(self, image_path, boot_action):
         then compares it against boot_action to determine whether to proceed with install.
         """
 
-        # All devices support cold boot
-        supported_boot = ['cold']
-
         # Verify image path exists
         if not os.path.exists(image_path):
             # Invalid image path
             return FW_AUTO_ERR_IMAGE
 
         # Check if post_install reboot is required
         try:
-            if self.get_firmware_update_notification(image_path) is None:
-                # No power cycle required
-                supported_boot += ['warm', 'fast', 'none', 'any']
-        except RuntimeError:
-            # Unknown error from firmware probe
-            return FW_AUTO_ERR_UKNOWN
-
-        if boot_action in supported_boot:
-            if self.update_firmware(image_path):
-                # Successful update
-                return FW_AUTO_INSTALLED
-            # Failed update (unknown reason)
-            return FW_AUTO_ERR_UKNOWN
+            reboot_required = self.get_firmware_update_notification(image_path) is not None
+        except RuntimeError as e:
+            return FW_AUTO_ERR_UKNOWN                    
+        
+        # Update if no reboot needed
+        if not reboot_required:
+            self.update_firmware(image_path)
+            return FW_AUTO_UPDATED
 
         # boot_type did not match (skip)
-        return FW_AUTO_ERR_BOOT_TYPE
+        if boot_action != "cold":
```

Comment:
Not sure I understand your question. The branch before checks if if a reboot is required to upgrade the SSD, if a reboot is not required then it installs and returns.

If a reboot is required we need to check that a reboot has actually been specified to the automatic firmware update command, else we need to fail out because we require a reboot and an insufficient reboot was provided.

Reply:
@alexrallen understood